### PR TITLE
Update/#31

### PR DESCRIPTION
**Pull request by YuraPunchyk**

### CODE REVIEW

File: src/main/java/com/chnu/controller/ProfileController.java [hunk 1]
```diff
@@ -19,27 +20,39 @@ public ProfileController(IProfileService profileService) {
...
+
+        if(object != null) {
+            return GenericResponse.of(profileService.update(object));
+        }
+        return GenericResponse.error("Profile is null.");
     }
 
     @DeleteMapping("")
-    public void delete(@RequestBody Profile object) {
+    public GenericResponse<Void> delete(@RequestBody Profile object) {
```
Comment (MikeMalyar) [inline src/main/java/com/chnu/controller/ProfileController.java]: This method may be removed.

Comment (MikeMalyar) [reply src/main/java/com/chnu/controller/ProfileController.java]: We can use the next one.

File: src/main/resources/database.properties [hunk 2]
```diff
@@ -4,7 +4,7 @@ postgresql.user=postgres
 postgresql.password=root
 
 hibernate.show_sql=true
-hibernate.hbm2ddl.auto=update
+hibernate.hbm2ddl.auto=create
```
Comment (MikeMalyar) [inline src/main/resources/database.properties]: Please revert this change.

File: src/main/java/com/chnu/controller/ProfileController.java [hunk 3]
```diff
@@ -19,27 +20,39 @@ public ProfileController(IProfileService profileService) {
     }
 
     @PostMapping("")
-    public Profile save(@RequestBody Profile profile) {
-        return profileService.save(profile).orElse(null);
+    public GenericResponse<Profile> save(@RequestBody Profile profile) {
+        GenericResponse<Profile> response = GenericResponse
+                .of(profileService.save(profile).orElse(null));
+        return response.getResult() != null ? response :
+                GenericResponse.error("This profile already exist.");
```
Comment (MikeMalyar) [inline src/main/java/com/chnu/controller/ProfileController.java]: Move this message into message properties and get it using properties utility.

File: src/main/java/com/chnu/controller/ProfileController.java [hunk 4]
```diff
@@ -19,27 +20,39 @@ public ProfileController(IProfileService profileService) {
...
     }
 
     @GetMapping("/{id}")
-    public Profile findById(@PathVariable(name = "id", required = true) Long pk) {
-        return profileService.findById(pk).orElse(null);
+    public GenericResponse<Profile> findById(@PathVariable(name = "id", required = true) Long pk) {
+        GenericResponse<Profile> response = GenericResponse
+                .of(profileService.findById(pk).orElse(null));
+        return response.getResult() != null ? response :
+                GenericResponse.error("Profile not found with id " + pk);
```
Comment (MikeMalyar) [inline src/main/java/com/chnu/controller/ProfileController.java]: Move this message into message properties and get it using properties utility.

File: src/main/java/com/chnu/controller/ProfileController.java [hunk 5]
```diff
@@ -19,27 +20,39 @@ public ProfileController(IProfileService profileService) {
...
 
     @PutMapping("")
-    public Profile update(@RequestBody Profile object) {
-        return profileService.update(object);
+    public GenericResponse<Profile> update(@RequestBody Profile object) {
+
+        if(object != null) {
+            return GenericResponse.of(profileService.update(object));
+        }
+        return GenericResponse.error("Profile is null.");
```
Comment (MikeMalyar) [inline src/main/java/com/chnu/controller/ProfileController.java]: Move this message into message properties and get it using properties utility.

File: src/main/java/com/chnu/controller/ProfileController.java [hunk 6]
```diff
@@ -19,27 +20,39 @@ public ProfileController(IProfileService profileService) {
...
+    public GenericResponse<Void> delete(@RequestBody Profile object) {
         profileService.delete(object);
+        return GenericResponse.empty();
     }
 
     @DeleteMapping("/{id}")
-    public void deleteById(@PathVariable(name = "id", required = true) Long pk) {
+    public GenericResponse<Void> deleteById(@PathVariable(name = "id", required = true) Long pk) {
         profileService.deleteById(pk);
+        return GenericResponse.empty();
```
Comment (MikeMalyar) [inline src/main/java/com/chnu/controller/ProfileController.java:54]: Maybe we need a message about successful object deleting ?

**Review by MikeMalyar — CHANGES_REQUESTED** [review]

**Review by MikeMalyar — APPROVED** [review]

**Review by ViktorPelepiak — APPROVED** [review]

**Review by lenyuk1999oo — APPROVED** [review]

**Pull request merged**